xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -432,6 +463,83 @@ int pick() {
     }
   }
 
+  // TODO: add javadocs comments
+  @VisibleForTesting
+  static final class StaticStrideScheduler {
+    private Vector<Long> scaledWeights;
```

Comment:
This should probably be an array, to avoid all the boxing of Long vs long. (I'm not trying to say whether this should be be 64 bit or 32 bit.)